jdbc-adapter/README.md

```diff
@@ -10,6 +10,8 @@ The JDBC adapter currently supports the following SQL dialects and data sources.
 * Oracle
 * Teradata
 * Redshift
+* [DB2](doc/db2-dialect.md)
```

Comment:
The other dialects are documented inline on the same page, so it might happen that in future we move the content to this page, or create separate pages for each dialect, to be consistent. For now, consistency is not so important as long as you can find things, so you are free to keep it a separate file, or move it directly to the README - we will merge the pull request anyway.